apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -16,17 +23,34 @@ export class ExternalToolUc {
 
 	async createExternalTool(userId: EntityId, externalToolCreate: ExternalToolCreate): Promise<ExternalTool> {
 		const externalTool = new ExternalTool({ ...externalToolCreate });
+		await this.addLogoToExternalToolIfExists(externalTool);
 
 		await this.ensurePermission(userId, Permission.TOOL_ADMIN);
 		await this.toolValidationService.validateCreate(externalTool);
 
-		const tool: Promise<ExternalTool> = this.externalToolService.createExternalTool(externalTool);
+		const tool: ExternalTool = await this.externalToolService.createExternalTool(externalTool);
 
 		return tool;
 	}
 
+	private async addLogoToExternalToolIfExists(externalTool: Partial<ExternalTool>): Promise<Partial<ExternalTool>> {
+		if (!externalTool.logoUrl) {
+			return externalTool;
+		}
+
+		const base64Logo: string | null = await this.externalToolService.fetchBase64Logo(externalTool.logoUrl);
+		if (base64Logo) {
+			externalTool.logoBase64 = base64Logo;
+		}
+
+		return externalTool;
```

Comment:
this is a function with side effects, so you are never using its return value, so it should return void.